### ISSUE TITLE
[release/2.14] 10 models up to 25% eager performance regression

### ISSUE THREAD

**Issue by bjarzemb** [intel/torch-xpu-ops#4924]
### 🐛 Describe the bug

Issue found during 2.14 RC test.

There is performance regression on Windows with Intel XPU (Arc B580). Batch size used: 2

Testcase | Model | Inductor status | Inductor diff | Inductor ref   latency [ms] | Inductor cur   latency [ms] | Eager status | Eager diff | Eager ref   latency [ms] | Eager cur   latency [ms]
-- | -- | -- | -- | -- | -- | -- | -- | -- | --
huggingface_training_bfloat16 | BartForCausalLM | PASS | -3.20% | 71.07 | 68.81 | FAIL | 10.70% | 76.06 | 84.18
huggingface_training_float16 | BartForCausalLM | PASS | -4.40% | 70.38 | 67.3 | FAIL | 10.70% | 73.76 | 81.63
huggingface_training_bfloat16 | GPT2ForSequenceClassification | PASS | -11.80% | 47.72 | 42.11 | FAIL | 9.50% | 63.83 | 69.93
huggingface_training_float16 | GPT2ForSequenceClassification | PASS | -11.60% | 47.26 | 41.77 | FAIL | 9.90% | 62.17 | 68.33
huggingface_training_bfloat16 | MBartForCausalLM | PASS | -4.30% | 71.88 | 68.78 | FAIL | 10.30% | 76.24 | 84.12
huggingface_training_float16 | MBartForCausalLM | PASS | -5.00% | 70.79 | 67.28 | FAIL | 10.30% | 73.94 | 81.56
huggingface_training_amp-bf16 | OPTForCausalLM | PASS | 0.10% | 127.76 | 127.91 | FAIL | 11.80% | 163.4 | 182.63
huggingface_training_amp-fp16 | OPTForCausalLM | PASS | -0.40% | 120.48 | 120.04 | FAIL | 12.10% | 152.07 | 170.47
huggingface_training_bfloat16 | OPTForCausalLM | PASS | 0.30% | 120.7 | 121.06 | FAIL | 16.00% | 141.53 | 164.21
huggingface_training_float16 | OPTForCausalLM | PASS | -0.90% | 115.43 | 114.37 | FAIL | 16.40% | 131.3 | 152.82
timm_models_training_amp-fp16 | res2next50 | PASS | 3.40% | 30.78 | 31.84 | FAIL | 8.00% | 33.49 | 36.16
torchbench_training_amp-fp16 | timm_vision_transformer_large | PASS | -6.50% | 1235.41 | 1155.68 | FAIL | 23.00% | 1506.89 | 1853.59
torchbench_training_float16 | timm_vision_transformer_large | PASS | 1.40% | 114.2 | 115.84 | FAIL | 7.50% | 152.04 | 163.4
timm_models_training_amp-bf16 | vit_base_patch14_dinov2.lvd142m | PASS | -4.30% | 84.02 | 80.41 | FAIL | 9.10% | 95.3 | 103.98
timm_models_training_amp-fp16 | vit_base_patch14_dinov2.lvd142m | PASS | -3.70% | 80.14 | 77.15 | FAIL | 9.60% | 91.96 | 100.75
timm_models_training_bfloat16 | vit_base_patch14_dinov2.lvd142m | PASS | -5.10% | 79.32 | 75.28 | FAIL | 14.60% | 80.3 | 92.02
timm_models_training_float16 | vit_base_patch14_dinov2.lvd142m | PASS | -2.80% | 74.53 | 72.41 | FAIL | 15.50% | 76.65 | 88.53
timm_models_training_float32 | vit_base_patch14_dinov2.lvd142m | PASS | 0.90% | 202.78 | 204.58 | FAIL | 8.30% | 212.24 | 229.76
timm_models_training_amp-fp16 | vit_base_patch16_siglip_256 | PASS | 3.40% | 26.96 | 27.87 | FAIL | 7.40% | 26.12 | 28.04
torchbench_training_bfloat16 | hf_Bart | PASS | 1.10% | 77.67 | 78.53 | FAIL | 7.30% | 86.92 | 93.25
torchbench_training_bfloat16 | hf_T5_large | PASS | 6.40% | 701.24 | 746.08 | FAIL | 11.20% | 1194.3 | 1328.53
torchbench_training_float32 | hf_Whisper | PASS | 0.50% | 32.56 | 32.72 | FAIL | 11.20% | 32.9 | 36.58

### Versions

<details>
<summary>Click to expand versions</summary>
PyTorch version: 2.14.0+xpu
Is debug build: False
CUDA used to build PyTorch: None
ROCM used to build PyTorch: N/A

OS: Microsoft Windows 11 Pro (10.0.26200 64-bit)
GCC version: Could not collect
Clang version: Could not collect
CMake version: version 3.31.6
Libc version: N/A

Python version: 3.12.13 | packaged by conda-forge | (main, Aug 11 2026, 10:18:27) [MSC v.1944 64 bit (AMD64)] (64-bit runtime)
Python platform: Windows-11-10.0.26200-SP0
Is CUDA available: False
CUDA runtime version: No CUDA
CUDA_MODULE_LOADING set to: N/A
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA
Is XPU available: True
XPU used to build PyTorch: 20260100
Intel GPU driver version:
* 32.0.101.8864 (20260717000000.******+***)
Intel GPU models onboard:
* Intel(R) Arc(TM) B580 Graphics
Intel GPU models detected:
* [0] _XpuDeviceProperties(name='Intel(R) Arc(TM) B580 Graphics', platform_name='Intel(R) oneAPI Unified Runtime over Level-Zero V2', type='gpu', device_id=0xE20B, uuid=86800be2-0000-0000-0300-000000000000, driver_version='1.15.38308', total_memory=11875MB, local_mem_size=128KB, last_level_cache_size=18432KB, max_compute_units=160, memory_clock_rate=0MHz, memory_bus_width=64-bit, gpu_eu_count=160, gpu_subslice_count=20, max_work_group_size=1024, max_num_sub_groups=64, sub_group_sizes=[16 32], has_fp16=1, has_fp64=1, has_atomic64=1, is_integrated_gpu=0)
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: False
Caching allocator config: N/A

CPU:
Name: 13th Gen Intel(R) Core(TM) i5-13400
Manufacturer: GenuineIntel
Family: 205
Architecture: 9
ProcessorType: 3
DeviceID: CPU0
CurrentClockSpeed: 2500
MaxClockSpeed: 2500
L2CacheSize: 9728
L2CacheSpeed: None
Revision: None

Versions of relevant libraries:
[pip3] dpcpp-cpp-rt==2026.1.0
[pip3] intel-cmplr-lib-rt==2026.1.0
[pip3] intel-cmplr-lib-ur==2026.1.0
[pip3] intel-cmplr-lic-rt==2026.1.0
[pip3] intel-opencl-rt==2026.1.0
[pip3] intel-openmp==2026.1.0
[pip3] intel-pti==1.0.1
[pip3] intel-sycl-rt==2026.1.0
[pip3] mkl==2026.1.0
[pip3] mkl-include==2024.2.0
[pip3] mkl-static==2024.2.0
[pip3] mypy_extensions==1.1.0
[pip3] numpy==1.26.4
[pip3] onemkl-license==2026.1.0
[pip3] onemkl-sycl-blas==2026.1.0
[pip3] onemkl-sycl-dft==2026.1.0
[pip3] onemkl-sycl-lapack==2026.1.0
[pip3] onemkl-sycl-rng==2026.1.0
[pip3] onemkl-sycl-sparse==2026.1.0
[pip3] onnx==1.21.0
[pip3] onnx-ir==0.1.16
[pip3] onnxscript==0.6.2
[pip3] optree==0.13.0
[pip3] tbb==2023.1.0
[pip3] tbb-devel==2021.13.1
[pip3] tcmlib==1.5.0
[pip3] torch==2.14.0+xpu
[pip3] torchaudio==2.11.0+xpu
[pip3] torchvision==0.29.0+xpu
[pip3] triton-xpu==3.8.0
[pip3] umf==1.1.0
[conda] dpcpp-cpp-rt                 2026.1.0         pypi_0              pypi
[conda] intel-cmplr-lib-rt           2026.1.0         pypi_0              pypi
[conda] intel-cmplr-lib-ur           2026.1.0         pypi_0              pypi
[conda] intel-cmplr-lic-rt           2026.1.0         pypi_0              pypi
[conda] intel-opencl-rt              2026.1.0         pypi_0              pypi
[conda] intel-openmp                 2026.1.0         pypi_0              pypi
[conda] intel-pti                    1.0.1            pypi_0              pypi
[conda] intel-sycl-rt                2026.1.0         pypi_0              pypi
[conda] mkl                          2026.1.0         pypi_0              pypi
[conda] mkl-include                  2024.2.0         pypi_0              pypi
[conda] mkl-static                   2024.2.0         pypi_0              pypi
[conda] numpy                        1.26.4           pypi_0              pypi
[conda] onemkl-license               2026.1.0         pypi_0              pypi
[conda] onemkl-sycl-blas             2026.1.0         pypi_0              pypi
[conda] onemkl-sycl-dft              2026.1.0         pypi_0              pypi
[conda] onemkl-sycl-lapack           2026.1.0         pypi_0              pypi
[conda] onemkl-sycl-rng              2026.1.0         pypi_0              pypi
[conda] onemkl-sycl-sparse           2026.1.0         pypi_0              pypi
[conda] optree                       0.13.0           pypi_0              pypi
[conda] tbb                          2023.1.0         pypi_0              pypi
[conda] tbb-devel                    2021.13.1        pypi_0              pypi
[conda] tcmlib                       1.5.0            pypi_0              pypi
[conda] torch                        2.14.0+xpu       pypi_0              pypi
[conda] torchaudio                   2.11.0+xpu       pypi_0              pypi
[conda] torchvision                  0.29.0+xpu       pypi_0              pypi
[conda] triton-xpu                   3.8.0            pypi_0              pypi
[conda] umf                          1.1.0            pypi_0              pypi

</details>

**Comment by daisyden** [comment]
<details>
<summary>label-issue: intel/torch-xpu-ops#4924</summary>

Root cause: Eager-path performance regression on Windows / Arc B580 in the 2.14 RC across many HF/timm/torchbench models (eager latency up 7-23% while Inductor path passes); representative BartForCausalLM bfloat16 training eager +10.7%.

Trace mode: evidence-only from issue body (per-model eager latency deltas).

Analyzed case: BartForCausalLM bfloat16 training (row 1 of 25).

| axis | value | reason |
|---|---|---|
| `type` | `Bug` | `issue_type` is `Bug`. |
| `test` | `test: e2e` | dynamo E2E benchmarks (huggingface/timm/torchbench), eager path. |
| `module` | `module: ops` | Eager-path aten regression (Inductor path passes, eager fails). |
| `priority` | `Urgent` | Measured perf regression >7% (eager deltas up to 23%). |
| `os` | `os: Windows` | Reported as performance regression on Windows (Windows 11, MSC build). |
| `hw` | `hw: BMG` | Arc B580 (device_id 0xE20B) is Battlemage. |
| `dtype` | `dtype: bfloat16` | Representative case is bfloat16 training. |
| symptom | `performance` | Eager latency regression. |
| symptom | `regression` | 2.14 RC vs prior ref latency baseline. |
| symptom | `training` | Representative case is training mode. |

</details>

**Comment by daisyden** [comment]
<details>
<summary>label-issue: intel/torch-xpu-ops#4924</summary>

Root cause: Eager-path performance regression on Windows / Arc B580 in the 2.14 RC across many HF/timm/torchbench models (eager latency up 7-23% while Inductor path passes); representative BartForCausalLM bfloat16 training eager +10.7%.

Trace mode: evidence-only from issue body (per-model eager latency deltas).

Analyzed case: BartForCausalLM bfloat16 training (row 1 of 25).

| axis | value | reason |
|---|---|---|
| `type` | `Bug` | `issue_type` is `Bug`. |
| `test` | `test: e2e` | dynamo E2E benchmarks (huggingface/timm/torchbench), eager path. |
| `module` | `module: ops` | Eager-path aten regression (Inductor path passes, eager fails). |
| `priority` | `Urgent` | Measured perf regression >7% (eager deltas up to 23%). |
| `os` | `os: Windows` | Reported as performance regression on Windows (Windows 11, MSC build). |
| `hw` | `hw: BMG` | Arc B580 (device_id 0xE20B) is Battlemage. |
| `dtype` | `dtype: bfloat16` | Representative case is bfloat16 training. |
| symptom | `performance` | Eager latency regression. |
| symptom | `regression` | 2.14 RC vs prior ref latency baseline. |
| symptom | `training` | Representative case is training mode. |

</details>

**Comment by bjarzemb** [comment]
BlenderbotSmall moved to seperate issue: https://github.com/intel/torch-xpu-ops/issues/5150

**Comment by bjarzemb** [comment]
torchbench_training_float16 hf_T5_large perf regression moved to seperate issue: https://github.com/intel/torch-xpu-ops/issues/5155

**Comment by bjarzemb** [comment]
Performance restored for all models in this issue with new nightly. See details below:

Suite | DT | Model | pt2.13 | 20260624 | 20260720 | 20260731 | 20260805 | 20260811 | pt2.14_rc | 20260823 | 20260825 | diff     (pt2.13 vs latest)
-- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | --
hf | amp-bf16 | OPTForCausalLM | 165.04 | 189.49 | 189.66 | 183.77 | 184.54 | 184.48 | 184.59 | 160.58 | 160.58 | -3%
huggingface | amp-fp16 | OPTForCausalLM | 153.64 | 177.72 | 177.99 | 172.31 | 172.31 | 172.24 | 172.02 | 148.33 | 148.28 | -3%
huggingface | bfloat16 | GPT2ForSequenceClassification | 65.13 | 71.18 | 71.13 | 71.12 | 70.99 | 70.95 | 70.96 | 64.43 | 64.44 | -1%
huggingface | bfloat16 | OPTForCausalLM | 143.23 | 167.08 | 167.43 | 165.77 | 165.65 | 165.82 | 165.97 | 141.88 | 141.99 | -1%
huggingface | bfloat16 | BartForCausalLM | 77.15 | 85.62 | 85.46 | 85.32 | 85.35 | 85.18 | 85.25 | 77.46 | 76.57 | -1%
huggingface | bfloat16 | MBartForCausalLM | 77.44 | 85.68 | 85.56 | 85.38 | 85.5 | 85.19 | 85.25 | 76.77 | 76.79 | -1%
huggingface | float16 | GPT2ForSequenceClassification | 63.25 | 69.54 | 69.55 | 69.61 | 69.57 | 69.43 | 69.46 | 62.85 | 62.86 | -1%
huggingface | float16 | OPTForCausalLM | 133.47 | 156.78 | 157.1 | 155.17 | 155.21 | 155.35 | 154.9 | 131.82 | 131.7 | -1%
huggingface | float16 | BartForCausalLM | 74.88 | 83.18 | 83.11 | 82.82 | 82.91 | 82.84 | 82.75 | 74.28 | 74.18 | -1%
huggingface | float16 | MBartForCausalLM | 75.17 | 83.21 | 83.11 | 82.97 | 82.92 | 82.68 | 82.73 | 74.28 | 74.46 | -1%
timm | amp-bf16 | vit_base_patch14_dinov2.lvd142m | 96 | 107.74 | 108.03 | 104.54 | 104.51 | 104.57 | 104.42 | 91.81 | 91.78 | -4%
timm | amp-fp16 | res2next50 | 66.46 |   | 66.53 | 66.14 | 67.47 | 67.54 |   | 61.87 | 62.35 | -6%
timm | amp-fp16 | vit_base_patch14_dinov2.lvd142m | 92.73 | 104.75 | 104.68 | 101.39 | 101.49 | 101.39 | 101.3 | 88.87 | 88.75 | -4%
timm | amp-fp16 | vit_base_patch16_siglip_256 | 49.73 | 47.74 | 50.62 | 49.06 | 51.16 | 48.85 | 51.07 | 44.85 | 45.33 | -9%
timm | bfloat16 | vit_base_patch14_dinov2.lvd142m | 81.09 | 93.68 | 92.65 | 92.42 | 92.58 | 92.51 | 92.65 | 80.78 | 80.81 | 0%
timm | float16 | vit_base_patch14_dinov2.lvd142m | 77.27 | 89.17 | 88.95 | 89.11 | 89.18 | 89.04 | 89.03 | 77.19 | 77.25 | 0%
timm | float32 | vit_base_patch14_dinov2.lvd142m | 212.51 | 230.16 | 230.63 | 229.94 | 230.07 | 229.96 | 230.25 | 213.48 | 214 | 1%
torchbench | amp-fp16 | timm_vision_transformer_large | 1970.42 |   |   |   | 2013.07 |   |   | 1896.45 | 1938.37 | -2%
torchbench | bfloat16 | hf_Bart | 71.35 |   |   |   | 72.82 |   |   | 65.41 | 66.15 | -7%
torchbench | bfloat16 | hf_T5_large | 1500.3 |   |   |   | 1563.71 |   |   | 1485.81 | 1549.89 | 3%
torchbench | float16 | timm_vision_transformer_large | 118.47 |   |   |   | 126.59 |   |   | 115.89 | 110.46 | -7%
torchbench | float32 | hf_Whisper | 32.4 |   |   |   | 35.96 |   |   | 32.23 | 32.2 | -1%

**Comment by fmazurx** [comment]
Restored performence confimed with package: https://download.pytorch.org/whl/test/xpu/torch-2.14.0%2Bxpu-cp312-cp312-win_amd64.whl#sha256=9acd595fe85000de53edcb6e0b176bb963b0eac8cf3239caf70b720041433fcb
